obsreport/obsreport_receiver.go

```diff
@@ -163,23 +246,11 @@ func (rec *Receiver) endOp(
 	span := trace.SpanFromContext(receiverCtx)
 
 	if rec.level != configtelemetry.LevelNone {
-		var acceptedMeasure, refusedMeasure *stats.Int64Measure
-		switch dataType {
-		case config.TracesDataType:
-			acceptedMeasure = obsmetrics.ReceiverAcceptedSpans
-			refusedMeasure = obsmetrics.ReceiverRefusedSpans
-		case config.MetricsDataType:
-			acceptedMeasure = obsmetrics.ReceiverAcceptedMetricPoints
-			refusedMeasure = obsmetrics.ReceiverRefusedMetricPoints
-		case config.LogsDataType:
-			acceptedMeasure = obsmetrics.ReceiverAcceptedLogRecords
-			refusedMeasure = obsmetrics.ReceiverRefusedLogRecords
+		if rec.useOtelForMetrics {
```

Comment:
Maybe hide this behind a `rec.record()` method that does the switching.

Reply:
Created a `recordMetrics` to hide it and also to not be mistaken with the `if span.IsRecording() ` line below which is not refer a different type of "record"